packages/creator-hub/renderer/src/hooks/useSceneCustomCode.ts

```diff
@@ -0,0 +1,46 @@
+import { useState, useCallback } from 'react';
+import { workspace } from '#preload';
+import { hasCustomCode } from '/shared/scene-parser';
+import type { Project } from '/shared/types/projects';
+
+interface UseSceneCustomCodeResult {
+  isCustomCode: boolean;
+  isLoading: boolean;
+  error: Error | null;
+  detectCustomCode: () => Promise<boolean>;
+}
+
+export function useSceneCustomCode(
```

Comment:
there is a custom code detection implementation in two different places with two different approaches:

- CH: uses `useSceneCustomCode` hook => calls `workspace.getSceneSourceFile()` => uses shared `hasCustomCode()`
- Inspector: calls scene RPC client => `getSceneCustomCode()` => also uses shared `hasCustomCode()`

this makes `packages/inspector/src/components/Renderer/Metrics/Metrics.tsx:168-182` to load on component mount, but creates a race condition

is it possible for the inspector to receive this information from the CH to avoid re-computing?